File: star-syncplayer/src/main/java/top/tonydon/util/observer/Observable.java
package top.tonydon.util.observer;

/**
 * 被观察者接口
 */
public interface Observable<T> {
    /**
     * 添加观察者
     *
     * @param observer 观察者对象
     */
    void addObserver(T observer);

    /**
     * 删除观察者
     *
     * @param observer 观察者对象
     */
    void removeObserver(T observer);
}

File: star-syncplayer-client/src/main/java/top/tonydon/syncplayer/util/observer/Observable.kt
package top.tonydon.syncplayer.util.observer

/**
 * 被观察者接口
 */
interface Observable<T> {
    /**
     * 添加观察者
     *
     * @param observer 观察者对象
     */
    fun addObserver(observer: T)

    /**
     * 删除观察者
     *
     * @param observer 观察者对象
     */
    fun removeObserver(observer: T)

}
